CHANGELOG.md

```diff
@@ -43,10 +43,32 @@ We also updated the shortcut to open a new terminal on macOS to Control + Shift
 See [configuration](docs/pages/connect-your-client/teleport-connect.mdx#configuration)
 for the current list of shortcuts.
 
-### Teleport plugins improvement
+#### Removal of support for Debian 9 and Ubuntu non-LTS versions
+
+Upon release of v16, we will no longer provide support or releases for the
+following Linux distro versions:
+* Debian 9 (stretch)
+* Ubuntu 16.10 (yakkety)
+* Ubuntu 17.04 (zesty)
+* Ubuntu 17.10 (artful)
+* Ubuntu 18.10 (cosmic)
+* Ubuntu 19.04 (disco)
+* Ubuntu 19.10 (eoan)
+* Ubuntu 20.10 (groovy)
+* Ubuntu 21.04 (hirsuite)
+* Ubuntu 21.10 (impish)
+* Ubuntu 22.10 (kinetic)
+* Ubuntu 23.04 (lunar)
+* Ubuntu 23.10 (mantic)
+
+This change applies to all Teleport versions.
```

Comment:
I think we should remove this support only starting from v16. Let's keep older versions publishing as-is, otherwise it will be unexpected breaking change.

Reply:
> I think we should remove this support only starting from v16. Let's keep older versions publishing as-is, otherwise it will be unexpected breaking change.

This is _really_ non trivial to do and takes the technical implementation from a 10 line `rm` PR to weeks of work, and likely other major breakage along the way. Do we still want to do this?

Reply:
Is that because the publishing tool is not versioned with Teleport? In that case, I don't think we should do this right now, we can't just stop publishing to a distro in a patch release. Longer-term, we should version publishing tool with Teleport, this is the direction we're moving to for all our tools/products anyway.

Reply:
> Is that because the publishing tool is not versioned with Teleport?

Largely because of this.

> Longer-term, we should version publishing tool with Teleport, this is the direction we're moving to for all our tools/products anyway.

I _really_ don't like this approach, as it makes maintenance much harder. IMO we should be pulling tools out of Teleport/e repos, and making them configurable enough to be invoked from Teleport workflows.

Either way though, it's unlikely that we can do this with our current OS package publishing approach as it's tightly coupled with some GH features that are default-branch dependent.